contracts/new_StakeVault.sol

```diff
@@ -27,17 +31,85 @@ contract StakeVault is Ownable {
 
     mapping(uint256 => PoolInfo) poolsById;
 
-    function addPool (uint256 poolId, address sponsor, uint256 expiration) public {}
+    function setPoolTracker(LaunchPoolTrackerTest launchPoolTracker) public {
+        _poolTrackerContract = launchPoolTracker;
+    }
+
+    modifier senderOwnsStake(uint256 stakeId) {
+        Stake memory st = _stakes[stakeId];
+        require(
+            st.staker == msg.sender,
+            "Account not authorized to unstake"
```

Comment:
"Account not authorized to unstake" -> Investor account not authorized to interact with the the specified Stake